File: library/src/main/java/io/ashdavies/rx/rxtasks/SingleEmitterSuccessListener.java
package io.ashdavies.rx.rxtasks;

import com.google.android.gms.tasks.OnSuccessListener;
import io.reactivex.SingleEmitter;
import io.reactivex.annotations.Nullable;

class SingleEmitterSuccessListener<T> implements OnSuccessListener<T> {

  private final SingleEmitter<T> emitter;

  SingleEmitterSuccessListener(SingleEmitter<T> emitter) {
    this.emitter = emitter;
  }

  @Override
  public void onSuccess(@Nullable T t) {
    if (t == null) {
      emitter.onError(new NullPointerException());
    }

    emitter.onSuccess(t);
  }
}

File: library/src/main/kotlin/io/ashdavies/rx/rxtasks/SingleEmitterSuccessListener.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.OnSuccessListener
import io.reactivex.SingleEmitter

internal class SingleEmitterSuccessListener<T>(private val emitter: SingleEmitter<T>) : OnSuccessListener<T> {

  override fun onSuccess(t: T?) {
    if (t == null) {
      emitter.onError(NullPointerException())
      return
    }

    emitter.onSuccess(t)
  }
}
